src/main/java/org/folio/consortia/service/impl/SharingInstanceServiceImpl.java

```diff
@@ -118,6 +128,48 @@ public SharingInstanceCollection getSharingInstances(UUID consortiumId, UUID ins
     return result;
   }
 
+  @Override
+  @Transactional
+  public void completePromotingLocalInstance(String eventPayload) {
+    log.debug("completePromotingLocalInstance:: parameters eventPayload: {}", eventPayload);
+    try {
+      var promotingEvent = objectMapper.readValue(eventPayload, SharingInstance.class);
+
+      String centralTenantId = tenantService.getCentralTenantId();
+      String sourceTenantId = promotingEvent.getSourceTenantId();
+      String targetTenantId = promotingEvent.getTargetTenantId();
+      checkTenantsExistAndContainCentralTenantOrThrow(sourceTenantId, targetTenantId);
+
+      if (ObjectUtils.notEqual(centralTenantId, targetTenantId)) {
+        log.warn("completePromotingLocalInstance:: promotion failed as targetTenantId: {} does not equal to centralTenantId: {}", targetTenantId, centralTenantId);
+        return;
+      }
+
+      var specification = constructSpecification(promotingEvent.getInstanceIdentifier(), sourceTenantId, targetTenantId, null);
+      var optionalSharingInstance = sharingInstanceRepository.findOne(specification);
+
+      if (optionalSharingInstance.isEmpty()) {
+        log.warn("completePromotingLocalInstance:: sharingInstance with instanceIdentifier: {}, sourceTenantId: {}, targetTenantId: {} does not exist",
+          promotingEvent.getInstanceIdentifier(), sourceTenantId, targetTenantId);
+        return;
+      }
+
+      var promotedSharingInstance = optionalSharingInstance.get();
+      if (ObjectUtils.isNotEmpty(promotingEvent.getError())) {
+        promotedSharingInstance.setStatus(Status.ERROR);
+        promotedSharingInstance.setError(promotingEvent.getError());
+      } else {
+        promotedSharingInstance.setStatus(Status.COMPLETE);
+      }
+
+      sharingInstanceRepository.save(promotedSharingInstance);
+      log.info("completePromotingLocalInstance:: status of sharingInstance with instanceIdentifier: {}, sourceTenantId: {}, targetTenantId: {} " +
+        "has been updated to: {}", promotedSharingInstance.getInstanceId(), sourceTenantId, targetTenantId, promotedSharingInstance.getStatus());
+    } catch (Exception e) {
+      log.error("completePromotingLocalInstance:: exception occurred while promoting local sharing instance", e);
```

Comment:
the same with populating error in sharing_instance table

Reply:
new SharingInstance() - in this case we will create new record in DB, better to revert this and remain only log if we could not catch

Reply:
Fixed